ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/StatsUpdater.java

```diff
@@ -81,10 +83,14 @@ public void gatherStats(HiveConf hiveConf, CompactionInfo ci, Map<String, String
                 sb.append(")");
             }
             sb.append(" compute statistics");
-            if (!conf.getBoolVar(HiveConf.ConfVars.HIVE_STATS_AUTOGATHER) && ci.isMajorCompaction()) {
-                List<String> columnList = msc.findColumnsWithStats(CompactionInfo.compactionInfoToStruct(ci));
+            if (ci.isMajorCompaction()) {
+                List<String> columnList = msc.findColumnsWithStats(CompactionInfo.compactionInfoToStruct(ci)).stream()
+                        .filter(columnName -> !StatsSetupConst.areColumnStatsUptoDate(tableProperties, columnName))
```

Comment:
maybe
````
boolean isStatsUpToDate = .. `columnList.stream.allMatch(columnName -> StatsSetupConst.areColumnStatsUptoDate(tableProperties, columnName))`
````

Reply:
> Maybe use allMatch
With the filter
```
.filter(columnName -> !StatsSetupConst.areColumnStatsUptoDate(tableProperties, columnName))
```
only the columns without stats collected and updated.

> Would it work for the partitioned table ?
Thanks for mentioning this; added partition support.

Reply:
> only the columns without stats collected and updated.

oh, I missed that